### PR TITLE
[zh] Sync blog 2023-08-23-kubelet-podresources-api-ga.md

### PR DESCRIPTION
**Pull request by wilsonwu**
Sync blog 2023-08-23-kubelet-podresources-api-ga.md

### CODE REVIEW

File: content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+In order for the containerized monitoring application consume the API, the socket should be mounted inside the container. A good practice is to mount the directory on which the podresources socket endpoint sits rather than the socket directly. This will ensure that after a kubelet restart, the containerized monitor application will be able to re-connect to the socket.
+-->
+为了让容器化监控应用程序使用 API，套接字应安装在容器内。
+一个好的做法是挂载 podresources 套接字端点所在的目录，而不是直接挂载套接字。
+这将确保 kubelet 重新启动后，容器化监视器应用程序将能够重新连接到套接字。
+
+<!--
+An example manifest for a hypothetical monitoring agent consuming the podresources API and deployed as a DaemonSet could look like:
+-->
+假设的监控代理使用 podresources API 并部署为 DaemonSet 的示例清单可能如下所示：
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md:107]:
```suggestion
在下面的 DaemonSet 的示例清单中，包含一个假象的使用 podresources API 的监控代理：
```

File: content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+      - hostPath:
+          path: /var/lib/kubelet/pod-resources
+          type: Directory
+        name: host-podresources
+```
+
+<!--
+I hope you find it straightforward to consume the podresources API  programmatically. The kubelet API package provides the protocol file and the go type definitions; however, a client package is not yet available from the project, and the existing code should not be used directly. The [recommended](https://github.com/kubernetes/kubernetes/blob/v1.28.0-rc.0/pkg/kubelet/apis/podresources/client.go#L32) approach is to reimplement the client in your projects, copying and pasting the related functions like for example the multus project is [doing](https://github.com/k8snetworkplumbingwg/multus-cni/blob/v4.0.2/pkg/kubeletclient/kubeletclient.go).
+-->
+我希望你发现以编程方式使用 podresources API 很简单。kubelet API包提供了协议文件和 go 类型定义；
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md:177]:
```suggestion
我希望你发现以编程方式使用 podresources API 很简单。kubelet API包提供了协议文件和 Go 类型定义；
```

File: content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+但是，该项目尚未提供客户端包，并且不应直接使用现有代码。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md:178]:
```suggestion
但是，该项目尚未提供客户端包，并且你也不应直接使用现有代码。
```

File: content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+[推荐](https://github.com/kubernetes/kubernetes/blob/v1.28.0-rc.0/pkg/kubelet/apis/podresources/client.go#L32)方法是在项目中重新实现客户端，
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md:179]:
```suggestion
[推荐](https://github.com/kubernetes/kubernetes/blob/v1.28.0-rc.0/pkg/kubelet/apis/podresources/client.go#L32)方法是在你自己的项目中重新实现客户端，
```

File: content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+复制并粘贴相关功能，例如 multus 项目正在[执行](https://github.com/k8snetworkplumbingwg/multus-cni/blob/v4.0.2/pkg/kubeletclient/kubeletclient.go)。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md:180]:
```suggestion
复制并粘贴相关功能，就像 multus 项目[所做的那样](https://github.com/k8snetworkplumbingwg/multus-cni/blob/v4.0.2/pkg/kubeletclient/kubeletclient.go)。
```

File: content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+-->
+在操作使用 podresources API 的容器化监控应用程序时，有几点值得强调，以防止出现“陷阱”：
+
+<!--
+- Even though the API only exposes data, and doesn't allow by design clients to mutate the kubelet state, the gRPC request/response model requires read-write access to the podresources API socket. In other words, it is not possible to limit the container mount to `ReadOnly`.
+- Multiple clients are allowed to connect to the podresources socket and consume the API, since it is stateless.
+- The kubelet has [built-in rate limits](https://github.com/kubernetes/kubernetes/pull/116459) to mitigate local Denial of Service attacks from misbehaving or malicious consumers. The consumers of the API must tolerate rate limit errors returned by the server. The rate limit is currently hardcoded and global, so misbehaving clients can consume all the quota and potentially starve correctly behaving clients.
+-->
+- 尽管 API 仅公开数据，并且设计上不允许客户端改变 kubelet 状态，
+  但 gRPC 请求/响应模型需要对 podresources API 套接字进行读写访问。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md:193]:
```suggestion
  但 gRPC 请求/响应模型要求能对 podresources API 套接字进行读写访问。
```

File: content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+  换句话说，不可能将容器挂载限制为 `ReadOnly`。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md:194]:
```suggestion
  换句话说，将容器挂载限制为 `ReadOnly` 是不可能的。
```

File: content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+- 允许多个客户端连接到 podresources 套接字并使用 API，因为它是无状态的。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md:195]:
```suggestion
- 让多个客户端连接到 podresources 套接字并使用此 API 是允许的，因为 API 是无状态的。
```

File: content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+- kubelet 具有[内置速率限制](https://github.com/kubernetes/kubernetes/pull/116459)，
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md:196]:
```suggestion
- kubelet 具有[内置限速机制](https://github.com/kubernetes/kubernetes/pull/116459)，
```

File: content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+  以减轻来自行为不当或恶意消费者的本地拒绝服务攻击。API 的使用者必须容忍服务器返回的速率限制错误。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md:197]:
```suggestion
  用以缓解来自行为不当或恶意用户的本地拒绝服务攻击。API 的使用者必须容忍服务器返回的速率限制错误。
```

File: content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+  速率限制目前是硬编码的和全局的，因此行为不当的客户端可能会消耗所有配额，并可能导致行为正确的客户端挨饿。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md:198]:
```suggestion
  速率限制目前是硬编码的且作用于全局的，因此行为不当的客户端可能会耗光所有配额，进而导致行为正确的客户端挨饿。
```

**Comment by windsonsea** [comment]
Hi @wilsonwu , you can use these commands to push all changes in one commit:

```
git commit --amend --no-edit
git push -f
```

Every reviewer mentioned here has received more than 30 spamed emails, which is triggered once you perform the push action.

File: content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md [hunk 12]
```diff
@@ -0,0 +1,231 @@
...
+podresources API 由本地 kubelet 提供，位于 kubelet 运行所在的同一节点上。
+在 Unix 风格的系统上，通过 Unix 域套接字提供端点；默认路径是 `/var/lib/kubelet/pod-resources/kubelet.sock`。
+在 Windows 上，通过命名管道提供端点；默认路径是 `npipe://\\.\pipe\kubelet-pod-resources`。
+
+<!--
+In order for the containerized monitoring application consume the API, the socket should be mounted inside the container. A good practice is to mount the directory on which the podresources socket endpoint sits rather than the socket directly. This will ensure that after a kubelet restart, the containerized monitor application will be able to re-connect to the socket.
+-->
+为了让容器化监控应用使用 API，套接字应挂载到容器内。
+一个好的做法是挂载 podresources 套接字端点所在的目录，而不是直接挂载套接字。
+这中做法将确保 kubelet 重新启动后，容器化监视器应用将能够重新连接到套接字。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md:102]:
```suggestion
这种做法将确保 kubelet 重新启动后，容器化监视器应用能够重新连接到套接字。
```

**Review by ydFu — APPROVED** [review]
/approve 
Thank

**Pull request merged**